__tests__/embed.ts

```diff
@@ -154,12 +127,138 @@ describe('embed.serlo.org/thumbnail?url=...', () => {
       expect(response.headers.get('content-type')).toBe('image/webp')
     })
 
-    test('returns placeholder when video does not exist', async () => {
-      const response = await requestThumbnail(
-        `https://player.vimeo.com/video/999999999?autoplay=1`
-      )
-      expect(isPlaceholderResponse(response))
+    describe('returns placeholder', () => {
+      test('when video does not exist', async () => {
+        const response = await requestThumbnail(
+          `https://player.vimeo.com/video/999999999?autoplay=1`
+        )
+        expect(isPlaceholderResponse(response))
+      })
+
+      test('when video id is malformed', async () => {
+        const response = await requestThumbnail(
+          `https://player.vimeo.com/video/foo:password@en.serlo.org/math`
+        )
+        expect(isPlaceholderResponse(response))
+      })
+
+      test('when request to vimeo api fails', async () => {
+        givenVimeoApi(hasInternalServerError())
+
+        const response = await requestThumbnail(
+          `https://player.vimeo.com/video/${video.id}?autoplay=1`,
+
+          TestEnvironment.Local
+        )
+        expect(isPlaceholderResponse(response))
+      })
+
+      test('when vimeo api returns malformed json', async () => {
+        givenVimeoApi(returnsMalformedJson())
+
+        const response = await requestThumbnail(
+          `https://player.vimeo.com/video/${video.id}?autoplay=1`,
+          TestEnvironment.Local
+        )
+        expect(isPlaceholderResponse(response))
+      })
+
+      describe('when vimeo api returns a malformed response', () => {
+        test.each([
+          42,
+          { type: 'video' },
+          { thumbnail_url: video.thumbnailUrl },
+        ])('%p', async (data) => {
+          givenVimeoApi(returnsJson(data))
+
+          const response = await requestThumbnail(
+            `https://player.vimeo.com/video/${video.id}?autoplay=1`,
+            TestEnvironment.Local
+          )
+          expect(isPlaceholderResponse(response))
+        })
+      })
+
+      test('when vimeo api returns a malformed thumbnail_url', async () => {
+        givenVimeoApi(returnsJson({ type: 'video', thumbnail_url: '42' }))
+
+        const response = await requestThumbnail(
+          `https://player.vimeo.com/video/${video.id}?autoplay=1`,
+          TestEnvironment.Local
+        )
+        expect(isPlaceholderResponse(response))
+      })
+
+      test('when vimeo api returns a thumbnail_url not pointing to vimeocdn.com', async () => {
```

Comment:
why are you testing the vimeo api here?

Reply:
Well… you are right.
But If the Vimeo-API is compromised so `vimeocdn.com` is  probably also compromised and they could serve us what they want.
Also this scenario is so unlikely that I personally prefer to have less tests, less code.

Reply:
Here I am interested what the opinion of @inyono is...